storage/postgres/client.go

```diff
@@ -297,3 +311,40 @@ func (c *Client) Wipe(ctx context.Context) error {
 
 	return nil
 }
+
+// DisableTriggersAndFKConstraints disables all triggers and foreign key constraints
+// in indexer tables. This is useful when inserting blockchain data out of order,
+// so that later blocks can refer to (yet unindexed) earlier blocks without violating constraints.
+func (c *Client) DisableTriggersAndFKConstraints(ctx context.Context) error {
+	// List all tables, then drop their triggers.
+	// FK constraints are implemented via triggers, so this also disables FK constraints.
+	tables, err := c.listIndexerTables(ctx)
+	if err != nil {
+		return err
+	}
+	for _, table := range tables {
+		c.logger.Info("disabling DB triggers on table", "table", table)
+		if _, err = c.pool.Exec(ctx, fmt.Sprintf("ALTER TABLE %s DISABLE TRIGGER ALL;", table)); err != nil {
```

Comment:
here and below: What do you think of batching all of these together so we never have an intermediate state where some tables have triggers and others don't?

In theory it's unnecessary since we're planning to only enable/disable triggers only when no other analyzers are running... but it also seems relatively simple to add